src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -332,6 +346,7 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setAllowedTypes('choices', ['null', 'array', '\Traversable']);
         $resolver->setAllowedTypes('choice_translation_domain', ['null', 'bool', 'string']);
         $resolver->setAllowedTypes('choice_loader', ['null', 'Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface', ChoiceLoader::class]);
+        $resolver->setAllowedTypes('choice_filter', ['null', 'callable', 'string', PropertyPath::class, ChoiceFilter::class]);
```

Comment:
Really as a bonus: could it support an array we'll convert to a `in_array` callable?
Referring to this [old comment of mine](https://github.com/symfony/symfony/pull/18334#issuecomment-266278790)  👼

Reply:
Thanks for the review :). Actually after more thoughts about this I [commented here](https://github.com/symfony/symfony/pull/28624#discussion_r223858780).
I think we should better add support for this in intl types only, as suggested by Javier in https://github.com/symfony/symfony/pull/35456#issuecomment-578483902.
We could use the `choices` option for those and that would be a shortcut for defining the `choice_filter` as array of choices. I propose to do that I another PR when this one is merged, WDYT?

Reply:
Alright, that makes sense. DX-wise, it could have been useful for other types without requiring to write more code, but I'll live without it :)